src/attributes.h

```diff
@@ -19,4 +19,14 @@
 #  endif
 #endif
 
+#if defined(__clang__)
+#  if __has_attribute(lifetimebound)
+#    define LIFETIMEBOUND [[clang::lifetimebound]]
+#  else
+#    define LIFETIMEBOUND
+#  endif
+#else
+#  define LIFETIMEBOUND
+#endif
```

Comment:
Build warning (in clang 11.0, but not gcc 9.3)
```
In file included from ./util/system.h:17:
./attributes.h:30:7: warning: extra tokens at end of #endif directive [-Wextra-tokens]
#endif#if defined(__clang__)
      ^
      //
1 warning generated.
```
```
((HEAD detached at origin/pr/19387))$ clang --version
Debian clang version 11.0.0-++20200708111116+1f84ace3c72-1~exp1~20200708091733.3345 
Target: x86_64-pc-linux-gnu
```

Reply:
I don't understand what this is complaining about. Looks like some subtle whitespace thing, but it's all fine as far as I can tell. Why wouldn't it warn about the block above as well?

Reply:
Same. Maybe that version of clang is buggy if no one else got it.

Reply:
I can't hit this with clang 10.

Does it spew a million warnings for our includes? That would make sense, at least. But if this is the only one there must be a reason.

If this is a real clang bug, free to bisect and report upstream since you're building from git, so we don't hit this when 11 is released :)

Reply:
Installed the new clang nightly updates to this version, built, and the warning is gone. (Yay!)
```
$ clang --version
Debian clang version 11.0.0-++20200709111134+dc4a6f5db4f-1~exp1~20200709091753.3347

$ uname -a
Linux 5.7.0-1-amd64 #1 SMP Debian 5.7.6-1 (2020-06-24) x86_64 GNU/Linux
```